framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -3130,6 +3154,26 @@ protected void checkThrownExpression(ThrowTree tree) {
         }
     }
 
+    /**
+     * Returns a set of AnnotatedTypeMirror that is allowed throw exceptions for method.
+     *
+     * <p>Note: If the method declaration do not have any throw clauses, then return null.
+     *
+     * @param mtree MethodTree to look up throw expressions
+     * @return List of AnnotationTypeMirrors, one per exception in method signature, that contains
+     *     annotation and java expressions
+     */
+    protected List<AnnotatedTypeMirror> getExceptionList(MethodTree mtree) {
+        List<AnnotatedTypeMirror> ExceptionList = new ArrayList<>();
+        List<? extends ExpressionTree> ExceptionTrees = mtree.getThrows();
+        if (!ExceptionTrees.isEmpty()) {
+            for (ExpressionTree throwExpression : ExceptionTrees) {
+                ExceptionList.add(atypeFactory.getAnnotatedType(throwExpression));
+            }
+        }
+        return ExceptionList;
```

Comment:
This method always returns a new list - not null when there is no throws clause.
It is better to return an empty list instead of null anyways, so please fix the mismatch and return the canonical empty list, not a new list, when there is no throws clause.